hpobench/abstract_benchmark.py

```diff
@@ -131,17 +132,37 @@ def _check_and_cast_configuration(configuration: Union[Dict, ConfigSpace.Configu
                                       configuration_space: ConfigSpace.ConfigurationSpace) \
             -> ConfigSpace.Configuration:
         """ Helper-function to evaluate the given configuration.
-            Cast it to a ConfigSpace.Configuration and evaluate if it violates some boundaries.
+            Cast it to a ConfigSpace.Configuration and evaluate if it violates its boundaries.
+
+            Note:
+                We remove inactive parameters from the configuration. Inactive parameters are parameters that are not
```

Comment:
We remove inactive hyperparameters from the given configuration. Inactive hyperparameters are hyperparameters that are not relevant for a configuration, e.g. A is only relevant if B=1 and if B!=1 then A is inactive will be removed from the configuration if present.